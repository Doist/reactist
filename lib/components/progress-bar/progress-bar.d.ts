import * as React from 'react';
import './progress-bar.less';
type Props = {
    /** Additional css class applied to the progress bar. */
    className?: string;
    /** How much of the progress bar should be filled. Number between 0 and 100 inclusive. */
    fillPercentage?: number;
    /** Defines the human readable text alternative for assitive technologies. */
    'aria-valuetext'?: string;
};
declare function ProgressBar({ fillPercentage, className, 'aria-valuetext': ariaValuetext }: Props): React.JSX.Element;
declare namespace ProgressBar {
    var displayName: string;
}
export { ProgressBar };
