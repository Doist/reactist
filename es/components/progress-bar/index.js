import { ProgressBar } from './progress-bar.js'
export { ProgressBar as default } from './progress-bar.js'
//# sourceMappingURL=index.js.map
