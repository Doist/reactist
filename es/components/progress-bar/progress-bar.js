import * as React from 'react'
import { HiddenVisually } from '../../hidden-visually/hidden-visually.js'
import classNames from 'classnames'

function ProgressBar({ fillPercentage = 0, className, 'aria-valuetext': ariaValuetext }) {
    const finalClassName = classNames('reactist_progress_bar', className)
    const width = fillPercentage < 0 ? 0 : fillPercentage > 100 ? 100 : fillPercentage
    return /*#__PURE__*/ React.createElement(
        'div',
        {
            className: finalClassName,
        },
        /*#__PURE__*/ React.createElement('div', {
            className: 'inner',
            style: {
                width: width + '%',
            },
        }),
        /*#__PURE__*/ React.createElement(
            HiddenVisually,
            null,
            /*#__PURE__*/ React.createElement('progress', {
                value: width,
                max: 100,
                'aria-valuetext': ariaValuetext != null ? ariaValuetext : undefined,
            }),
        ),
    )
}

ProgressBar.displayName = 'ProgressBar'

export { ProgressBar }
//# sourceMappingURL=progress-bar.js.map
