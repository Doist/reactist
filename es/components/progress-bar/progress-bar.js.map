{"version":3,"file":"progress-bar.js","sources":["../../../src/components/progress-bar/progress-bar.tsx"],"sourcesContent":["import * as React from 'react'\nimport { HiddenVisually } from '../../hidden-visually'\nimport classNames from 'classnames'\n\nimport './progress-bar.less'\n\ntype Props = {\n    /** Additional css class applied to the progress bar. */\n    className?: string\n    /** How much of the progress bar should be filled. Number between 0 and 100 inclusive. */\n    fillPercentage?: number\n    /** Defines the human readable text alternative for assitive technologies. */\n    'aria-valuetext'?: string\n}\nfunction ProgressBar({ fillPercentage = 0, className, 'aria-valuetext': ariaValuetext }: Props) {\n    const finalClassName = classNames('reactist_progress_bar', className)\n    const width = fillPercentage < 0 ? 0 : fillPercentage > 100 ? 100 : fillPercentage\n    return (\n        <div className={finalClassName}>\n            <div className=\"inner\" style={{ width: `${width}%` }} />\n            <HiddenVisually>\n                <progress value={width} max={100} aria-valuetext={ariaValuetext ?? undefined} />\n            </HiddenVisually>\n        </div>\n    )\n}\nProgressBar.displayName = 'ProgressBar'\n\nexport { ProgressBar }\n"],"names":["ProgressBar","fillPercentage","className","ariaValuetext","finalClassName","classNames","width","React","createElement","style","HiddenVisually","value","max","undefined","displayName"],"mappings":";;;;AAcA,SAASA,WAAT,CAAqB;AAAEC,EAAAA,cAAc,GAAG,CAAnB;EAAsBC,SAAtB;EAAiC,gBAAkBC,EAAAA,aAAAA;AAAnD,CAArB,EAA8F;AAC1F,EAAA,MAAMC,cAAc,GAAGC,UAAU,CAAC,uBAAD,EAA0BH,SAA1B,CAAjC,CAAA;AACA,EAAA,MAAMI,KAAK,GAAGL,cAAc,GAAG,CAAjB,GAAqB,CAArB,GAAyBA,cAAc,GAAG,GAAjB,GAAuB,GAAvB,GAA6BA,cAApE,CAAA;AACA,EAAA,oBACIM,KAAA,CAAAC,aAAA,CAAA,KAAA,EAAA;AAAKN,IAAAA,SAAS,EAAEE,cAAAA;AAAhB,GAAA,eACIG,KAAA,CAAAC,aAAA,CAAA,KAAA,EAAA;AAAKN,IAAAA,SAAS,EAAC,OAAf;AAAuBO,IAAAA,KAAK,EAAE;AAAEH,MAAAA,KAAK,EAAKA,KAAL,GAAA,GAAA;AAAP,KAAA;AAA9B,GAAA,CADJ,eAEIC,KAAA,CAAAC,aAAA,CAACE,cAAD,EAAe,IAAf,eACIH,KAAA,CAAAC,aAAA,CAAA,UAAA,EAAA;AAAUG,IAAAA,KAAK,EAAEL,KAAjB;AAAwBM,IAAAA,GAAG,EAAE,GAA7B;IAAkD,gBAAAT,EAAAA,aAAA,IAAAA,IAAAA,GAAAA,aAAA,GAAiBU,SAAAA;GAAnE,CADJ,CAFJ,CADJ,CAAA;AAQH,CAAA;;AACDb,WAAW,CAACc,WAAZ,GAA0B,aAA1B;;;;"}