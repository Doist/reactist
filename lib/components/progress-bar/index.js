"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./progress-bar.js");exports.default=e.ProgressBar;
//# sourceMappingURL=index.js.map
