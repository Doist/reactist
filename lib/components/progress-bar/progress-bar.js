"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("react"),r=require("../../hidden-visually/hidden-visually.js"),t=require("classnames");function a(e){return e&&"object"==typeof e&&"default"in e?e:{default:e}}function l(e){if(e&&e.__esModule)return e;var r=Object.create(null);return e&&Object.keys(e).forEach((function(t){if("default"!==t){var a=Object.getOwnPropertyDescriptor(e,t);Object.defineProperty(r,t,a.get?a:{enumerable:!0,get:function(){return e[t]}})}})),r.default=e,r}var n=l(e),s=a(t);function u({fillPercentage:e=0,className:t,"aria-valuetext":a}){const l=s.default("reactist_progress_bar",t),u=e<0?0:e>100?100:e;return n.createElement("div",{className:l},n.createElement("div",{className:"inner",style:{width:u+"%"}}),n.createElement(r.HiddenVisually,null,n.createElement("progress",{value:u,max:100,"aria-valuetext":null!=a?a:void 0})))}u.displayName="ProgressBar",exports.ProgressBar=u;
//# sourceMappingURL=progress-bar.js.map
