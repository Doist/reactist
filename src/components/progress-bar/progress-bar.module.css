:root {
    --reactist-progressbar-height: var(--reactist-spacing-xsmall);
    --reactist-progressbar-radius-outer: var(--reactist-progressbar-height);
    --reactist-progressbar-radius-inner: var(--reactist-progressbar-height);
    --reactist-progressbar-track: var(--reactist-framework-fill-crest);
    --reactist-progressbar-fill: var(--reactist-bg-brand);
}

.progressBar {
    height: var(--reactist-progressbar-height);
    border-radius: var(--reactist-progressbar-radius-outer);
    background-color: var(--reactist-progressbar-track);
    width: 100%;
}

.inner {
    height: inherit;
    border-radius: var(--reactist-progressbar-radius-inner);
    background-color: var(--reactist-progressbar-fill);
}
