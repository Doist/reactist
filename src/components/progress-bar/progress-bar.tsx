import { HiddenVisually } from '../../hidden-visually'
import classNames from 'classnames'

import styles from './progress-bar.module.css'

type Props = {
    /** Additional css class applied to the progress bar. */
    className?: string
    /** How much of the progress bar should be filled. Number between 0 and 100 inclusive. */
    fillPercentage?: number
    /** Defines the human readable text alternative for assitive technologies. */
    'aria-valuetext'?: string
}
function ProgressBar({ fillPercentage = 0, className, 'aria-valuetext': ariaValuetext }: Props) {
    const finalClassName = classNames(styles.progressBar, className)
    const width = fillPercentage < 0 ? 0 : fillPercentage > 100 ? 100 : fillPercentage
    return (
        <div className={finalClassName}>
            <div className={styles.inner} style={{ width: `${width}%` }} />
            <HiddenVisually>
                <progress value={width} max={100} aria-valuetext={ariaValuetext ?? undefined} />
            </HiddenVisually>
        </div>
    )
}
ProgressBar.displayName = 'ProgressBar'

export { ProgressBar }
