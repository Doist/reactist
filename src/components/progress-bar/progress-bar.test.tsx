import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import { ProgressBar } from './progress-bar'

describe('ProgressBar', () => {
    it('renders without crashing', () => {
        const progressBar = render(<ProgressBar />)
        expect(progressBar.container).toMatchSnapshot()
    })

    it('uses 0% width for fillPercentages smaller than 0', () => {
        render(<ProgressBar fillPercentage={-1} />)
        expect(screen.getByRole('progressbar')).toHaveValue(0)
    })

    it('uses 100% width for fillPercentages larger than 100', () => {
        render(<ProgressBar fillPercentage={1337} />)
        expect(screen.getByRole('progressbar')).toHaveValue(100)
    })

    describe('a11y', () => {
        it('renders with no a11y violations', async () => {
            const { container } = render(<ProgressBar fillPercentage={50} />)
            const results = await axe(container)

            expect(results).toHaveNoViolations()
        })

        it('supports the `aria-valuetext` attribute', () => {
            render(<ProgressBar fillPercentage={50} aria-valuetext="Step 2: Copying files..." />)
            expect(screen.getByRole('progressbar')).toHaveAttribute(
                'aria-valuetext',
                'Step 2: Copying files...',
            )
        })
    })
})
