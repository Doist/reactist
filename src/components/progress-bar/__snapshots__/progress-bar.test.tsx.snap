// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders without crashing 1`] = `
<div>
  <div
    class="reactist_progress_bar"
  >
    <div
      class="inner"
      style="width: 0%;"
    />
    <div
      class="hiddenVisually box position-absolute overflow-hidden"
    >
      <progress
        max="100"
        value="0"
      />
    </div>
  </div>
</div>
`;
